### PR TITLE
Add  Folly 2021.08.30.00

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **folly/2021.08.30.00**

The latest version available on Conan Center is 1 year older and [Examples](https://github.com/conan-io/examples) is outdated.

No new component/library have been added since 1 year ago, only the required C++ standard is now C++17.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/folly/all/conanfile.py [hunk 1]
```diff
@@ -45,41 +51,26 @@ def config_options(self):
...
 
     def requirements(self):
-        self.requires("boost/1.75.0")
+        self.requires("boost/1.76.0")
         self.requires("bzip2/1.0.8")
         self.requires("double-conversion/3.1.5")
         self.requires("gflags/2.2.2")
         self.requires("glog/0.4.0")
         self.requires("libevent/2.1.12")
+        self.requires("openssl/1.1.1k")
```
Comment (uilianries) [inline recipes/folly/all/conanfile.py:69]:
```suggestion
        self.requires("openssl/1.1.1l")
```

**Comment by uilianries** [comment]
depends on Libevent, which requires an outdated openssl version

**Comment by prince-chrismc** [comment]
Is there anything I can do to unblock this?

**Comment by uilianries** [comment]
Not about [this](https://github.com/conan-io/conan-center-index/pull/7128#issuecomment-959872753) error:

```
Undefined symbols for architecture x86_64:
  "___cxa_increment_exception_refcount", referenced from:
      folly::exception_ptr_get_type(std::exception_ptr const&) in libfolly.a(Exception.cpp.o)
      folly::exception_ptr_get_object(std::exception_ptr const&, std::type_info const*) in libfolly.a(Exception.cpp.o)
```

Looking into Facebook projects, they said they don't support M1. 
I'll skip Clang 12 and 2021.xxx for now because I can see a lot of other complains on Folly issues and don't want to spend more time finding more patches.

**Comment by sourcedelica** [comment]
Should this be reopened?

**Comment by uilianries** [comment]
Nope

**Comment by sourcedelica** [comment]
If you don't mind me asking, where does the PR stand currently?  I see a comment about libevent requiring an outdated openssl - is that the blocker?  Thanks!

**Comment by sourcedelica** [comment]
Thanks - we don't use Macos anyway, only Linux and Windows.  I'll try to build your PR branch for Linux and Windows.

**Comment by sourcedelica** [comment]
Hey @uilianries - 

Based on your bincrafters:folly/2021.08.30.00 branch I updated to Folly v2022.01.31.00 and got it building across Windows, Linux and Mac.  I don't have an M1 box though, so I wasn't able to test that.

Please see my PR to your PR :) at https://github.com/bincrafters/conan-center-index/pull/44.

**Comment by uilianries** [comment]
@sourcedelica Thank you! Let's see if we obtain some luck now.

File: recipes/folly/config.yml [hunk 2]
```diff
@@ -3,3 +3,5 @@ versions:
       folder: all
     "2020.08.10.00":
       folder: all
+    "2021.07.20.01":
```
Comment (uilianries) [inline recipes/folly/config.yml:6]:
```suggestion
    "2022.01.31.00":
```

**Comment by sourcedelica** [comment]
It looks like there are no AppleClang packages in CCI for Folly currently. Should we say it's only supported from 2022.01.31.00 forward?

**Comment by uilianries** [comment]
@sourcedelica Yes, as it's 2 years old that older package.

**Comment by sourcedelica** [comment]
Added in another PR to bincrafters:folly/2021.08.30.00 - https://github.com/conan-io/conan-center-index/pull/9479.

**Comment by sourcedelica** [comment]
https://github.com/bincrafters/conan-center-index/pull/47 to [bincrafters:folly/2021.08.30.00](https://github.com/bincrafters/conan-center-index/tree/folly/2021.08.30.00) removes apple-clang support for < 2022.01.31.00

**Comment by uilianries** [comment]
@sourcedelica The best approach would be closing this PR and open a new open, using a updated version. As we know, Folly lives on the edge and each tag can be quite different. This PR is many months old for a project that change many things in short intervals.

**Comment by sourcedelica** [comment]
An updated version of Folly?  We've tested with 2022.01.31.00 and that works.

**Comment by prince-chrismc** [comment]
Given how hard each version is I think sticking with what we got is a safe choice 😆

**Comment by uilianries** [comment]
> Given how hard each version is I think sticking with what we got is a safe choice 😆

Yes. When I tried to update, I tested different versions until finding one with only few errors 😂

**Comment by uilianries** [comment]
I don't get, it merged from master, but I didn't send any commit/push, only opened conflict page 👁️

**Comment by ericLemanissier** [comment]
Actually I think this PR can be closed, now that  https://github.com/conan-io/conan-center-index/pull/9479 is merged

**Pull request closed without merging**